src/main/java/org/elasticsearch/action/support/replication/TransportShardReplicationOperationAction.java

```diff
@@ -623,31 +645,41 @@ void performOnReplica(final ReplicationState state, final ShardRouting shard, fi
                 // to wait until they get the new mapping through the cluster
                 // state, which is why we recommend pre-defined mappings for
                 // indices using shadow replicas
-                state.onReplicaSuccess();
+                state.onReplicaSuccess(newReplicaResponseInstance());
```

Comment:
I think we need a different solution for shadow replicas instead of an empty response. This will be solved with #10749 but can we add a no commit here?